lite/kernels/xpu/flatten_compute.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "lite/kernels/xpu/flatten_compute.h"
+#include <vector>
+#include "lite/backends/xpu/xpu_header_sitter.h"
+#include "lite/core/op_registry.h"
+
+namespace paddle {
+namespace lite {
+namespace kernels {
+namespace xpu {
+
+void FlattenContiguousRangeCompute::Run() {
+  auto& param = Param<operators::FlattenContiguousRangeParam>();
+  auto x = param.x;
+  auto out = param.out;
+  auto out_dims = out->dims();
+  auto out_lod = out->lod();
+  // out->CopyDataFrom(*x);
+  out->Resize(out_dims);
+  out->set_lod(out_lod);
+  out->ShareDataWith(*x);
```

Comment:
```suggestion
  out->ShareDataWith(*x);
  out->Resize(out_dims);
```
ShareDataWith中做了dims的共享，把Resize方后面。
lod和x是一致的，在ShareDataWith中有处理，不需要额外set

Reply:
Done